File: Domasna_2/prototip/stocktopus-backend/src/main/java/mk/finki/ukim/mk/stocktopusbackend/service/FavoriteStocksService.java
package mk.finki.ukim.mk.stocktopusbackend.service;

import mk.finki.ukim.mk.stocktopusbackend.model.Stock;
import mk.finki.ukim.mk.stocktopusbackend.model.dto.StockDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface FavoriteStocksService {
    void addFavoriteStock(String username, Long stockId);

    void removeFavoriteStock(String username, Long stockId);

    Page<Stock> getFavoriteStocks(String username, Pageable pageable);
}

File: Domasna_2/prototip/stocktopus-frontend/src/service/favoriteStocksService.ts
import axiosInstance from '../config/axiosInstance';
import { StockDTO } from "../model/dto/stockDTO.ts";
import config from "../config/config.ts";
import {FavoriteStocksDTO} from "../model/dto/FavoriteStocksDTO.ts";
const BASE_URL = config.API_BASE_URL;
interface PaginationParams {
    page: number;
    size: number;
}

export const getFavoriteStocks = async ({ username, page, size }: { username: string } & PaginationParams) => {
    try {
        const response = await axiosInstance.get<{
            totalElements: number;
            content: StockDTO[];
        }>(`${BASE_URL}/favorite-stocks`, {
            params: {
                username,
                page,
                size
            },
        });
        return response.data;
    } catch (error) {
        console.error("Error fetching favorite stocks:", error);
        throw error;
    }
};
export const addFavoriteStock = async (data: FavoriteStocksDTO) => {
    await axiosInstance.post(`${BASE_URL}/favorite-stocks/add`, data);
};
export const removeFavoriteStock = async (data: FavoriteStocksDTO) => {
    await axiosInstance.delete(`${BASE_URL}/favorite-stocks/remove`, { data });
};
